### PR TITLE
Support OIDC config in cluster shell

### PR DESCRIPTION
**Pull request by colin014**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
- Support OIDC config download in `cluster shell`
- Support OIDC config download in `cluster config` command
- Refactor `login` command

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/pkg/browser v0.0.0-20180916011732-0a3d74bf9ce4
 	github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35 // indirect
 	github.com/sirupsen/logrus v1.4.2
-	github.com/spf13/afero v1.2.1 // indirect
+	github.com/skratchdot/open-golang v0.0.0-20190104022628-a2dfa6d0dab6
```
Comment (bonifaido) [inline go.mod:25]: This library is not used in the code.

File: internal/cli/command/login/oidc.go [hunk 2]
```diff
@@ -77,284 +35,11 @@ func runServer(banzaiCli cli.Cli, pipelineBasePath string) (string, error) {
 	}
 	issuerURL.Path = "/dex"
```
Comment (bonifaido) [inline internal/cli/command/login/oidc.go:36]: We could get the issuerURL more correctly here with calling `/auth/dex/login` on `pipelineBasePath` and check the redirect `Location` header.

Comment (colin014) [reply internal/cli/command/login/oidc.go]: cool thx, i added this

Comment (bonifaido) [reply internal/cli/command/login/oidc.go]: great!

**Pull request merged**